hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -357,7 +367,7 @@ private RetryDecision shouldRetry(final IOException ioe, final int retryCount,
     // check for the case of cluster unavailable state
     if (isClusterUnAvailable(nsId)) {
       // we allow to retry once if cluster is unavailable
-      if (retryCount == 0) {
```

Comment:
This logic starts with this counter at a value and then decreases, it makes sense to keep it as such.

Reply:
In fact, not all clients are so smart, they may not configred appropriately. When those clients encounter this problem, they complain about router not retry more.